app/controllers/favorites_controller.rb

```diff
@@ -4,10 +4,20 @@ class FavoritesController < ApplicationController
   # GET /favorites or /favorites.json
   def index
     @favorites = Favorite.all
+    @user_transactions = UserTransaction.all
+    @credit_cards = []
+    @favorites.each do |favorite|
+      @credit_cards.append(CreditCard.find_by(id: favorite.credit_card_id))
```

Comment:
good logic here to add the CreditCard by ID and append it directly to the creditcards array to use later, very handy